src/raw/cgroupv2.go

```diff
@@ -0,0 +1,146 @@
+}
+
+// NewCgroupsV2Fetcher creates a new cgroups data fetcher.
+func NewCgroupsV2Fetcher(hostRoot string, systemCPUReader SystemCPUReader, cgroupDriver string) (*CgroupsV2Fetcher, error) {
+	return &CgroupsV2Fetcher{
+		cgroupDriver:    cgroupDriver,
+		hostRoot:        hostRoot,
+		systemCPUReader: systemCPUReader,
+	}, nil
+}
+
+// Fetch get the metrics that can be found in cgroups v2 file system
+// Unlike v1, cgroup v2 has only single hierarchy.
+//TODO: populate also network from libcgroups
+func (cg *CgroupsV2Fetcher) Fetch(c types.ContainerJSON) (Metrics, error) {
+	stats := Metrics{}
+
+	pid := c.State.Pid
+	containerID := c.ID
+
+	var (
+		cgroupInfo *cgroupV2Paths
+		err        error
+	)
+
+	cgroupInfo, err = getCgroupV2Paths(cg.hostRoot, pid, cg.cgroupDriver, containerID)
+	if err != nil {
+		return stats, err
+	}
+
+	manager, err := cgroupsV2.LoadManager(cgroupInfo.MountPoint, cgroupInfo.Group)
+	if err != nil {
+		return stats, err
+	}
+
+	metrics, err := manager.Stat()
+	if err != nil {
+		return stats, err
+	}
+
+	stats.Time = time.Now()
+
+	if stats.Pids, err = cg.pids(metrics); err != nil {
+		log.Error("couldn't read pids stats: %v", err)
+	}
+
+	if stats.CPU, err = cg.cpu(metrics); err != nil {
+		log.Error("couldn't read cpu stats: %v", err)
+	}
+
+	if stats.CPU.Shares, err = getSingleFileUintStat(cgroupInfo, "cpu.weight"); err != nil {
+		log.Error("couldn't read cpu weight: %v", err)
+	}
+
+	if stats.Memory, err = cg.memory(metrics); err != nil {
+		log.Error("couldn't read memory stats: %v", err)
+	}
+
+	stats.ContainerID = containerID
+
+	netMetricsPath := filepath.Join(cg.hostRoot, "/proc", strconv.Itoa(pid), "net", "dev")
+	stats.Network, err = network(netMetricsPath)
+	if err != nil {
+		log.Error(
+			"couldn't fetch network stats for container %s from cgroups: %v",
+			containerID,
+			err,
+		)
+		return stats, err
+	}
+
+	return stats, nil
+}
+
+func (cg *CgroupsV2Fetcher) cpu(metric *cgroupstatsV2.Metrics) (CPU, error) {
+	if metric.CPU == nil || metric.CPU.UsageUsec == 0 {
+		return CPU{}, errors.New("no CPU metrics information")
+	}
+
+	cpu := CPU{
+		TotalUsage:        metric.CPU.UsageUsec * uint64(time.Microsecond),
+		UsageInUsermode:   metric.CPU.UserUsec * uint64(time.Microsecond),
+		UsageInKernelmode: metric.CPU.SystemUsec * uint64(time.Microsecond),
+	}
+
+	if metric.CPU.NrThrottled != 0 {
+		cpu.ThrottledPeriods = metric.CPU.NrThrottled
+		cpu.ThrottledTimeNS = metric.CPU.ThrottledUsec * uint64(time.Microsecond)
+	}
+
+	var err error
+	cpu.SystemUsage, err = cg.systemCPUReader.ReadUsage()
+
+	return cpu, err
+}
+
+func (cg *CgroupsV2Fetcher) memory(metric *cgroupstatsV2.Metrics) (Memory, error) {
+	mem := Memory{}
+	if metric.Memory == nil {
+		return mem, errors.New("no Memory metrics information")
+	}
+	if metric.Memory.Usage != 0 {
+		mem.UsageLimit = metric.Memory.UsageLimit
+		mem.FuzzUsage = metric.Memory.Usage
+	}
+
+	mem.Cache = metric.Memory.File
+	mem.RSS = metric.Memory.Anon
```

Comment:
In the table created to show the mapping of each value in NR, cgroupsV1 and cgroupsV2 
It's saying that the memvory.rss is memory.swap_limit I gues it's a typo on that table, isn't it? Because here I see the RSS metric comes from Memory.Anon

Reply:
Yes, it was actually a typo. Already updated, thanks!